phpseclib/File/X509.php

```diff
@@ -4614,4 +4614,27 @@ function _extractBER($str)
         $temp = preg_match('#^[a-zA-Z\d/+]*={0,2}$#', $temp) ? base64_decode($temp) : false;
         return $temp != false ? $temp : $str;
     }
+
+    /**
+     * Returns the OID corresponding to a name
+     *
+     * What's returned in the associative array is either a name or an OID if no OID to name mapping is available.
+     * The problem with this is that what may be an unmapped OID in one version of phpseclib may not be unmapped
+     * in the next version, so apps that are looking at this OID may not be able to work from version to version.
+     *
+     * This method will return the OID if a name is passed to it and if no mapping is avialable it'll assume that
+     * what's being passed to it already is an OID and return that instead. A few examples.
```

Comment:
I don't think the switch statements you proposed makes sense. Either you work on IDs, or you work on names. If you work on IDs, you don't have to translate to names. If you work on names, you don't handle IDs anyway, so returning an ID is pointless.

mapOID does what you implemented, getOID does what I described (return null/false if not found). Not sure this is consistent with existing methods though.